### PR TITLE
dbeaver/pro#1596 improve naming for objects outside defaults

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes dbeaver/pro#1596
Objects outside of defaults will use fully qualified name

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java [hunk 1]
```diff
@@ -388,19 +403,37 @@ protected String addDBMetadataToRequest(
         return additionalMetadata.toString();
     }
 
+    private boolean isRequiresFullyQualifiedName(@NotNull DBSObject object, @Nullable DBCExecutionContext context) {
+        if (context == null || context.getContextDefaults() == null
+        ) {
+            return true;
+        }
+        DBSObject parent = object.getParentObject();
+        return parent == null || !(
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java]: What is the reason to use a fully qualified name for an object without a parent? How this full name will be created?

Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java]:
```suggestion
        return parent != null && !(
```

File: plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java [hunk 2]
```diff
@@ -388,19 +403,37 @@ protected String addDBMetadataToRequest(
...
+    private boolean isRequiresFullyQualifiedName(@NotNull DBSObject object, @Nullable DBCExecutionContext context) {
+        if (context == null || context.getContextDefaults() == null
+        ) {
+            return true;
+        }
+        DBSObject parent = object.getParentObject();
+        return parent != null && !(
+            parent.equals(context.getContextDefaults().getDefaultCatalog())
+                || parent.equals(context.getContextDefaults().getDefaultSchema()));
+    }
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java]:
```suggestion
    private boolean isRequiresFullyQualifiedName(@NotNull DBSObject object, @Nullable DBCExecutionContext context) {
        if (context == null || context.getContextDefaults() == null) {
            return true;
        }
        DBSObject parent = object.getParentObject();
        DBSObject defaults= context.getContextDefaults();
        return parent != null && !(parent.equals(defaults.getDefaultCatalog()) || parent.equals(defaults.getDefaultSchema()));
    }
```

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java]: By the way, `context.getContextDefaults()` can be nullable

Comment (Destrolaric) [reply plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java]: context.getContextDefaults() == null?

**Review by LonwoLonwo — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java [hunk 3]
```diff
@@ -388,19 +403,37 @@ protected String addDBMetadataToRequest(
...
 
+    private boolean isRequiresFullyQualifiedName(@NotNull DBSObject object, @Nullable DBCExecutionContext context) {
+        if (context == null || context.getContextDefaults() == null) {
+            return false;
+        }
+        DBSObject parent = object.getParentObject();
+        DBCExecutionContextDefaults contextDefaults = context.getContextDefaults();
+        return parent != null && !(
+            parent.equals(contextDefaults.getDefaultCatalog())
+                || parent.equals(contextDefaults.getDefaultSchema()));
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java]:
```suggestion
        return parent != null && !(parent.equals(contextDefaults.getDefaultCatalog()) || parent.equals(contextDefaults.getDefaultSchema()));
```

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/gpt3/GPTCompletionEngine.java]: it fits into one line, but okk

**Review by E1izabeth — APPROVED** [review]

**Comment by MashaKorax** [comment]
verified

**Pull request merged**